tests/test_etc/test_docstring.py

```diff
@@ -0,0 +1,43 @@
+import pytest
+from xtgeo.xyz._xyz import inherit_docstring
+import inspect
+
+
+class SuperClass:
+    def my_function(self):
+        """I have a doc string"""
+
+
+def test_docstring():
+
+    class SubClass(SuperClass):
+        @inherit_docstring(inherit_from=SuperClass.my_function)
+        def my_function(self):
+            pass
+
+    assert SubClass.my_function.__doc__ == "I have a doc string"
+
+
+def test_new_docstring():
+
+    class SubClass(SuperClass):
+        @inherit_docstring(inherit_from=SuperClass.my_function)
+        def my_function(self):
+            """I have a different doc string"""
+            pass
+
+    assert SubClass.my_function.__doc__ == "I have a different doc string"
+
+
+def test_no_super_docstring():
+    class SuperClass:
+        def my_function(self):
+            pass
+
+    class SubClass(SuperClass):
+        @inherit_docstring(inherit_from=SuperClass.my_function)
+        def my_function(self):
+            """I have a doc string"""
+            pass
+
+    assert SubClass.my_function.__doc__ == "I have a doc string"
```

Comment:
I added a test documenting how the decorator will work. We could of course do other tings like concatenate if it is set on both functions, but that felt a bit overreaching. We could also make it simpler by always overwriting regardless.

Reply:
I think this looks good, as long as we make sure we don't simply overwrite docstrings on any functions that modifies behaviour (or changes default values, given that default values are specified in docstring)

Reply:
I\`ll write a comment in the decorator that we dont overwrite. We are only modifying the doc string, which has no impact on the default values. `help` is a concatenation of default values and doc string.

Reply:
>I'll write a comment in the decorator that we dont overwrite.

Ah, yeah, we're not overwriting docstrings.. 

>We are only modifying the doc string, which has no impact on the default values

I of course agree, but consider:

```
    class SuperClass:
        def my_function(self, myarg=5):
            """takes myarg (defaults to 5)"""
            pass

     class SubClass(SuperClass):
        @inherit_docstring(inherit_from=SuperClass.my_function)
        def my_function(self, myarg=10):
            super().my_function(myarg=myarg)
```

Here the docstring of the subclass would be incorrect. If the default values are not given in the docstring than this doesn't matter.

>help is a concatenation of default values and doc string.

Then never mind me :P

Reply:
> Here the docstring of the subclass would be incorrect. If the default values are not given in the docstring than this doesn't matter.

That is true, but we should trust ourselves enough to not do this ;)